### DISCUSSION TITLE
Control RG6 Gripper mounted on UR5 robot arm

### DISCUSSION THREAD

**Discussion by binhkhongbeo** [ros-industrial/universal_robot#660] [Q&A]
Hello there,
I would like to control the OnRobot RG6 gripper  that mounted to a UR5 robot arm via ROS noetic. But now I am having a problem in connecting my PC to robot through Ethernet.

UR5 was running just fine based on universal_robot's instruction (without RG6)
Then I tried to control RG6 based on instruction in the link: https://github.com/Osaka-University-Harada-Laboratory/onrobot but still use the Ethernet Connection & External Control script(URCap) like UR5 case.
Firstly, I assumed the gripper and the arm share the same IP so I use the robot IP, and it ran into error.
I keep trying by changing the robot IP to the same IP as default IP of RG6 Gripper, which is 192.168.1.1 and default port: 502, but nothing cool happen and eventually I got this ERROR which i could not fix :(

**ERROR**:

```
AttributeError: 'ExceptionResponse' object has no attribute 'registers'
[DEBUG] [1696923112.318935]: [/OnRobotRGInput] failed to receive incoming message : unable to receive data from sender, check sender's logs for details
[OnRobotRGTcpNode-2] process has died [pid 14410, exit code 1, cmd /home/mngu/catkin_ws/devel/lib/onrobot_rg_control/OnRobotRGTcpNode.py __name:=OnRobotRGTcpNode __log:=/home/mngu/.ros/log/a4614100-673e-11ee-9256-03bb5f0dd7a5/OnRobotRGTcpNode-2.log].
log file: /home/mngu/.ros/log/a4614100-673e-11ee-9256-03bb5f0dd7a5/OnRobotRGTcpNode-2*.log
[OnRobotRGTcpNode-2] restarting process
```

**Comment by gavanderhoorn** [comment]
It seems you already / also have https://github.com/UniversalRobots/Universal_Robots_ROS_Driver/issues/662 open. You also have https://github.com/Osaka-University-Harada-Laboratory/onrobot/issues/14 open.

Please don't cross-post.

I'm going to close this one, and give the maintainers of `Osaka-University-Harada-Laboratory/onrobot` some time to respond.
